airflow/utils/dot_renderer.py

```diff
@@ -42,6 +45,80 @@ def _refine_color(color: str):
     return color
 
 
+def _draw_nodes(node: TaskMixin, parent_graph: graphviz.Digraph, states_by_task_id: Dict[str, str]) -> None:
+    """Draw the node and its children on the given parent_graph recursively."""
+    if isinstance(node, BaseOperator):
+        # Draw task
+        if states_by_task_id:
+            state = states_by_task_id.get(node.task_id, State.NONE)
+            color = State.color_fg(state)
+            fill_color = State.color(state)
+        else:
+            color = node.ui_fgcolor
+            fill_color = node.ui_color
+
+        parent_graph.node(
+            node.task_id,
+            _attributes={
+                "label": node.label,
+                "shape": "rectangle",
+                "style": "filled,rounded",
+                "color": _refine_color(color),
+                "fillcolor": _refine_color(fill_color),
+            },
+        )
```

Comment:
How about introducing `_draw_task` and `_draw_task_group` to encapsulate the logic and split logic of `_draw_nodes` into smaller pieces?

Reply:
Thanks @turbaszek . Done